projects/packages/forms/src/contact-form/class-contact-form.php

```diff
@@ -733,11 +733,27 @@ private static function esc_shortcode_val( $val ) {
 	public static function parse_contact_field( $attributes, $content ) {
 		// Don't try to parse contact form fields if not inside a contact form
 		if ( ! Contact_Form_Plugin::$using_contact_form_field ) {
-			$att_strs = array();
+			$type = isset( $attributes['type'] ) ? $attributes['type'] : null;
```

Comment:
I think the issue starts around this process, it's our old "double escaping" friend. See how the fields are being parsed:

individual contact-field-option:
```
[07-Mar-2023 16:31:56 UTC] ATTRIBUTES: Array
(
    [label] => 5 - Excellent
    [type] => field-option
)

[07-Mar-2023 16:31:56 UTC] CONTENT:
[07-Mar-2023 16:31:56 UTC] OUTPUT: [contact-field-option label="5 - Excellent" type="field-option"/]
```

SC/MC contact-field:
```
[07-Mar-2023 16:31:56 UTC] ATTRIBUTES: Array
(
    [label] => Please rate our website
    [required] => 1
    [requiredText] => (required)
    [options] => Array
        (
        )

    [type] => radio
)

[07-Mar-2023 16:31:56 UTC] CONTENT:
[contact-field-option label="1 - Very Bad" type="field-option"/]

[contact-field-option label="2 - Poor" type="field-option"/]

[contact-field-option label="3 - Average" type="field-option"/]

[contact-field-option label="4 - Good" type="field-option"/]

[contact-field-option label="5 - Excellent" type="field-option"/]

[07-Mar-2023 16:31:56 UTC] OUTPUT: [contact-field label="Please rate our website" required="1" requiredText="(required)" options="1 - Very Bad,2 - Poor,3 - Average,4 - Good,5 - Excellent" type="radio"]
[contact-field-option label=&quot;1 - Very Bad&quot; type=&quot;field-option&quot;/]

[contact-field-option label=&quot;2 - Poor&quot; type=&quot;field-option&quot;/]

[contact-field-option label=&quot;3 - Average&quot; type=&quot;field-option&quot;/]

[contact-field-option label=&quot;4 - Good&quot; type=&quot;field-option&quot;/]

[contact-field-option label=&quot;5 - Excellent&quot; type=&quot;field-option&quot;/]
[/contact-field]
```

Some of the loops or calls is, apparently, messing with the content and escaping the fields. There must be a way to add an exception (yet not compromise security).

Reply:
The issue was happening because the Field Option block was being handled as the other forms fields and being added to the list of form fields.
After the submission, the page was generated based on this list of fields.
The implementation I'm using here is parsing the MC/SC content and generating the list of options used later the render the HTML.
That way, the content inside the MS/SC blocks doesn't really matter.

To fix the issue, I just removed the callback from the `contact-field-option` add_shortcode function: https://github.com/Automattic/jetpack/pull/29290/files#diff-673ad67b4f4d4ec951c7e85e5d273adaf51fd8a1b6029ed179260e2e7b716500R1042